esp-hal/src/uart.rs

```diff
@@ -1763,15 +1763,15 @@ mod asynch {
         T: Instance,
     {
         /// See [`UartRx::read_async`]
-        async fn read_async(&mut self, buf: &mut [u8]) -> Result<usize, Error> {
+        pub async fn read_async(&mut self, buf: &mut [u8]) -> Result<usize, Error> {
             self.rx.read_async(buf).await
         }
 
-        async fn write_async(&mut self, words: &[u8]) -> Result<usize, Error> {
+        pub async fn write_async(&mut self, words: &[u8]) -> Result<usize, Error> {
```

Comment:
I'm aware there are some undocumented public APIs now, however I am also working on revamping the documentation for this entire module, so I opted to just leave them out for now and take care of it in the other branch.